### ISSUE TITLE
Error upon interacting with quest-book

### ISSUE THREAD

**Issue by Worlfy** [ezeiger92/QuestWorld2#128]
Hi hi,

When trying to interact with my quest-book (/qe book), I get [this](https://hastebin.com/itemutivak.cs) error
`return p.getMetadata("questworld.last-object").stream().findFirst().map(metadata -> (DataObject) metadata.value()).orElse(null);` seems to be the cause

No way for me to open my quest-book atm, it just bugs out

Kind regards,
zThana

**Comment by ezeiger92** [comment]
That error is not possible under normal use, did you use /reload or something like Plugman to reload questworld?

**Comment by Worlfy** [comment]
Hmm, no. I'm mainly "against" plugin handlers such as Plugman. Restarting, reloading and using /qe reload all cause this to happen

**Comment by ezeiger92** [comment]
`java.lang.ClassCastException: com.questworld.quest.Quest cannot be cast to com.questworld.api.contract.DataObject`
The Quest class implements the DataObject interface, this exception is only possible if a different classloader was used. Every time this has popped up, it's been due to /reload.

Fully restart the server, and if the issue still exists send me the full server log file.

**Comment by Worlfy** [comment]
A full restart did the job. How come it breaks upon reloading?

**Comment by ezeiger92** [comment]
Reload unloads all the classes and recreates them with a brand new classloader. Metadata isn't cleared out, apparently, so the object still references the old class (and old classloader), and java can't cast between different classloaders as the classes could be vastly different. Rough description, anyway.

tl;dr QuestWorld is not reload safe.

**Comment by ezeiger92** [comment]
No, using /reload is bad practice to begin with and has been advised against for years and there is no reason that /reload would benefit questworld.
